Cubical/Structures/Group/MorphismProperties.agda

```diff
@@ -28,63 +28,84 @@ private
   variable
     ℓ ℓ' ℓ'' : Level
 
+infixr 31 _□_
+
+open Group
+open GroupHom
+open IsMonoid
+open IsSemigroup
+open IsGroup
+
 isPropIsGroupHom : (G : Group {ℓ}) (H : Group {ℓ'}) {f : ⟨ G ⟩ → ⟨ H ⟩} → isProp (isGroupHom G H f)
 isPropIsGroupHom G H {f} = isPropΠ2 λ a b → Group.is-set H _ _
 
 isSetGroupHom : {G : Group {ℓ}} {H : Group {ℓ'}} → isSet (GroupHom G H)
 isSetGroupHom {G = G} {H = H} = isOfHLevelRespectEquiv 2 equiv (isSetΣ (isSetΠ λ _ → is-set H) λ _ → isProp→isSet (isPropIsGroupHom G H)) where
-  open Group
-  equiv : (Σ[ g ∈ (Carrier G → Carrier H) ] (isGroupHom G H g)) ≃ GroupHom G H
-  equiv = isoToEquiv (iso (λ (g , m) → grouphom g m) (λ (grouphom g m) → g , m) (λ _ → refl) λ _ → refl)
+  equiv : (Σ[ g ∈ (Carrier G → Carrier H) ] (isGroupHom G H g)) ≃ (GroupHom G H)
+  equiv =  isoToEquiv (iso (λ (g , m) → grouphom g m) (λ hom → fun hom , isHom hom) (λ a → η-hom _) λ _ → refl)
 
 -- Morphism composition
 isGroupHomComp : {F : Group {ℓ}} {G : Group {ℓ'}} {H : Group {ℓ''}} →
-  (f : GroupHom F G) → (g : GroupHom G H) → isGroupHom F H (GroupHom.fun g ∘ GroupHom.fun f)
-isGroupHomComp (grouphom f morph-f) (grouphom g morph-g) x y =
-  cong g (morph-f _ _) ∙ morph-g _ _
+  (f : GroupHom F G) → (g : GroupHom G H) → isGroupHom F H (fun g ∘ fun f)
+isGroupHomComp f g x y = cong (fun g) (isHom f _ _) ∙ isHom g _ _
+
 
 compGroupHom : {F : Group {ℓ}} {G : Group {ℓ'}} {H : Group {ℓ''}} → GroupHom F G → GroupHom G H → GroupHom F H
-compGroupHom {F = F} {G = G} {H = H} f g = grouphom _ (isGroupHomComp f g)
+fun (compGroupHom f g) = fun g ∘ fun f
+isHom (compGroupHom f g) = isGroupHomComp f g
 
+open GroupEquiv
 compGroupEquiv : {F : Group {ℓ}} {G : Group {ℓ'}} {H : Group {ℓ''}} → GroupEquiv F G → GroupEquiv G H → GroupEquiv F H
-compGroupEquiv {F = F} {G = G} {H = H} f g =
-  groupequiv (compEquiv f.eq g.eq) (isGroupHomComp f.hom g.hom) where
-  module f = GroupEquiv f
-  module g = GroupEquiv g
+eq (compGroupEquiv f g) = compEquiv (eq f) (eq g)
+isHom (compGroupEquiv f g) = isHom (compGroupHom (hom f) (hom g))
+
+_□_ : _
+_□_ = compGroupEquiv
```

Comment:
Do you use this?

Reply:
Yes, in ZCohomology/Groups

Reply:
Can you just rename it locally in this file by importing the compGroupEquiv using this name? It's not good to pollute the global name space with notations like this